src/Nethermind/Nethermind.Mev/Source/BundlePool.cs

```diff
@@ -141,6 +160,52 @@ public bool AddBundle(MevBundle bundle)
             return false;
         }
 
+        public bool AddMegabundle(MevMegabundle megabundle)
+        {
+            Metrics.MegabundlesReceived++;
+            BundleEventArgs bundleEventArgs = new(megabundle, megabundle.RelaySignature);
+            NewReceived?.Invoke(this, bundleEventArgs);
+
+            if (ValidateBundle(megabundle))
+            {
+                Address relayAddress = _ecdsa.RecoverAddress(megabundle.RelaySignature, megabundle.Hash)!;
+                if (IsTrustedRelay(relayAddress))
+                {
+                    Metrics.ValidMegabundlesReceived++;
+                    NewPending?.Invoke(this, bundleEventArgs);
+
+                    _megabundles.AddOrUpdate(relayAddress,
+                        _ => megabundle,
+                        (_, _) =>
+                        {
+                            RemoveSimulation(megabundle);
```

Comment:
Remove simulation, not for bundle we are adding, but for bundle we are replacing (one of parameters of this lambda function)